src/web/mjs/connectors/Komiku.mjs

```diff
@@ -9,21 +9,31 @@ export default class Komiku extends WordPressMangastream {
         super.label = 'Komiku';
         this.tags = ['manga', 'indonesian'];
         this.url = 'https://komiku.co.id';
-        this.path = '/daftar-komik/';
+        this.path = '/manga/page/';
 
-        this.queryMangas = 'div#a-z ol li.ranking1 h4';
+        this.queryMangas = 'div.daftar div.bge div.kan > a.popunder';
         this.queryChapters = 'table.chapter tbody tr td.judulseries a';
         this.queryChaptersTitle = undefined;
         this.queryPages = 'div#Baca_Komik img[src]:not([src=""])';
     }
 
     async _getMangas() {
-        let request = new Request(new URL(this.path, this.url), this.requestOptions);
-        let data = await this.fetchDOM(request, this.queryMangas);
+        let mangaList = [];
+        for(let page = 1, run = true; run; page++) {
+            let mangas = await this._getMangasFromPage(page);
+            mangas.length === 20 ? mangaList.push(...mangas) : run = false;
```

Comment:
Does this work when the last page has only 17 mangas, will it still be added?
What happens if the last page always has 20 mangas which are repeated, should there be an additional abort limit such as `&& page < 100`?

Reply:
is there a way to handle 404 in Hakuneko other than throw an error to the user? [http://komiku.co.id/manga/page/92/](http://komiku.co.id/manga/page/92/)

Reply:
found solution end up using try and catch
```
    async _getMangasFromPage(page) {
        try {
            const uri = new URL(this.path + page + '/', this.url);
            const request = new Request(uri, this.requestOptions);
            const data = await this.fetchDOM(request, this.queryMangas);
            return data.map(element => {
                return {
                    id: this.getRootRelativeOrAbsoluteLink(element, request.url),
                    title: element.text.trim()
                };
            });
        } catch(error) {
            console.log(error);
            return [];
        }
    }
```